src/lib/transition_frontier_controller/transition_frontier_controller.ml

```diff
@@ -78,5 +82,12 @@ module Make (Inputs : Inputs_intf) :
       ~catchup_breadcrumbs_reader ;
     Catchup.run ~logger ~network ~frontier ~catchup_job_reader
       ~catchup_breadcrumbs_writer ;
+    Strict_pipe.Reader.iter clear_reader ~f:(fun _ ->
```

Comment:
we made `iter_sync` for this usecase where you just return `Deferred.unit` at the end

Reply:
It should work. `iter_without_pushback` is exactly what you need here, I'm pretty certain.